runtime/new-recipe.js

```diff
@@ -219,10 +219,8 @@ class ViewConnection extends Connection {
   toString(nameMap) {
     let result = [];
     result.push(this.name || '*');
-    if (['in', 'out', 'inout'].indexOf(this.direction) == -1)
-      result.push(`{${this.direction}}`)
-    else
-      result.push({'in': '<-', 'out': '->', 'inout': '='}[this.direction]);
+    // TODO: better deal with unspecified direction.
+    result.push({'in': '<-', 'out': '->', 'inout': '='}[this.direction] || '=');
```

Comment:
I had to change it so that it would roundtrip. The API setter should probably assert that we're setting it to one of these or `null`